core/compiler/src/main/scala/com/fortysevendeg/exercises/compiler/codeGen.scala

```diff
@@ -0,0 +1,142 @@
+package com.fortysevendeg.exercises
+package compiler
+
+import scala.language.implicitConversions
+import scala.reflect.api.Universe
+
+case class CodeGen[U <: Universe](override val u: U = scala.reflect.runtime.universe) extends TreeHelpers[U] {
+  import u._
+
+  def emitExercise(
+    name: Option[String]
+  ) = {
+    val term = makeTermName("EXC", name)
+    term → q"""
+      object $term extends Exercise {
+        override val name = $name
+      }
+      """
+  }
+
+  def emitSection(
+    name:          String,
+    description:   Option[String] = None,
+    exerciseTerms: List[TermName] = Nil
+  ) = {
+    val term = makeTermName("SEC", name)
+    term → q"""
+      object $term extends Section {
+        override val name         = $name
+        override val description  = $description
+        override val exercises    = $exerciseTerms
+      }
+      """
+  }
+
+  def emitLibrary(
+    name:         String,
+    description:  String,
+    color:        String,
+    sectionTerms: List[TermName] = Nil
+  ) = {
+    val term = makeTermName("LIB", name)
+    term → q"""
+      object $term extends Library {
+        override val name         = $name
+        override val description  = $description
+        override val color        = $color
+        override val sections     = $sectionTerms
+      }
+    """
+  }
+
+  def emitPackage(
+    packageName:   String,
+    libraryTree:   Tree,
+    sectionTrees:  List[Tree] = Nil,
+    exerciseTrees: List[Tree] = Nil
+  ) = q"""
+      package ${makeRefTree(packageName)} {
+        import com.fortysevendeg.exercises.Exercise
+        import com.fortysevendeg.exercises.Library
+        import com.fortysevendeg.exercises.Section
+
+        $libraryTree
+        ..$sectionTrees
+        ..$exerciseTrees
+      }
+    """
+
+}
+
+sealed trait TreeHelpers[U <: Universe] {
+  val u: U
+  import u._
+
+  protected def makeTermName(appellation: String, name: String): TermName =
+    makeTermName(appellation, Some(name))
+  protected def makeTermName(appellation: String, name: Option[String]): TermName =
```

Comment:
Empty lines between methods. We should also setup Codacy in this project. @fedefernandez Can help maybe